src/compiler/scala/tools/nsc/backend/jvm/BTypes.scala

```diff
@@ -56,22 +68,25 @@ abstract class BTypes {
     }
 
     final def buildString(builder: java.lang.StringBuilder): Unit = this match {
-      case UNIT   => builder.append('V')
-      case BOOL   => builder.append('Z')
-      case CHAR   => builder.append('C')
-      case BYTE   => builder.append('B')
-      case SHORT  => builder.append('S')
-      case INT    => builder.append('I')
-      case FLOAT  => builder.append('F')
-      case LONG   => builder.append('J')
-      case DOUBLE => builder.append('D')
-      case ClassBType(internalName) => builder.append('L').append(internalName).append(';')
-      case ArrayBType(component)    => builder.append('['); component.buildString(builder)
-      case MethodBType(args, res)   =>
-        builder.append('(')
-        args.foreach(_.buildString(builder))
-        builder.append(')')
-        res.buildString(builder)
+      case _: PrimitiveBType =>
+        this match {
+          case UNIT => builder.append('V')
```

Comment:
Did you consider using a HashMap for this lookup?

Reply:
The common case is `! PrimitiveBType`, so the main thing is to get to that case quickly. A reasonable and performant alternative would be to make it a virtual method.

Reply:
could make the descriptor a field of the PrimativeType

i.e. case p: PrimitiveBType => builder.append(p.descriptorChar)

Is that what you meant?